File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // initialize variables
    int coins = 0;
    float change;

    // prompts user for how much change they are owed
    do
    {
        change = get_float("Change owed:\n");
    }
    while (change < 0);

    // takes floating point input and converts and
    // rounds it for more accurate counting
    change = change * 100;
    change = round(change);

    // loop to detirmine fewest amount of coins necessary for change
    while (change > 0)
    {
        if (change >= 25)
        {
            change = change - 25;
            coins = coins + 1;
        }

        else if (change >= 10)
        {
            change = change - 10;
            coins = coins + 1;
        }

        else if (change >= 5)
        {
            change = change - 5;
            coins = coins + 1;
        }

        else
        {
            change = change - 1;
            coins = coins + 1;
        }
    }
    // prints fewest coins output
    printf("Fewest coins possible: %i\n", coins);
}

File: pset6/cash/cash.py
def cash():

    while True:
        try:
            # gets change owed to user and checks it for errors
            change = float(input("Change owed: "))
            # checks if input is greater than 0.0
            if change < 0.0:
                print("Please enter change owed.\n")
                continue
        # checks if input is a number
        except ValueError:
            print("I'm not sure what you mean...\n")
            continue
        else:
            break

    # takes floating point input and converts and
    # rounds it for more accurate counting
    change = change * 100
    change = int(round(change))

    # initializes number of coins to return to user
    coins = 0

    # checks for quarters
    while change >= 25:
        change = change - 25
        coins += 1

    # checks for dimes
    while change >= 10:
        change = change - 10
        coins += 1

    # checks for nickels
    while change >= 5:
        change = change - 5
        coins += 1

    # checks for pennies
    while change >= 1:
        change = change - 1
        coins += 1

    print("Fewest coins possible:", coins)


if __name__ == "__main__":
    cash()